ui/widgets/nuxeo-collapsible.js

```diff
@@ -0,0 +1,174 @@
+   *
+   * Example:
+   *
+   *     <nuxeo-collapsible chevron chevronAtStart expanded>
+   *        <label slot="heading">Nuxeo</label>
+   *        <div>Collapsed content</div>
+   *     </nuxeo-collapsible>
+   *
+   * ### Styling
+   * The following custom properties and mixins are available for styling:
+   * Custom property | Description | Default
+   * ----------------|-------------|----------
+   * `--nuxeo-collapsible-heading-padding` | Heading padding | `4px`
+   * `--nuxeo-collapsible-icon-height` | Chevron height | `20px`
+   * `--nuxeo-collapsible-icon-width` | Chevron width | `20px`
+   *
+   * @memberof Nuxeo
+   *
+   */
+
+  class Collapsible extends mixinBehaviors([], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <style>
+          button {
+            height: 100%;
+            width: 100%;
+            display: flex;
+            justify-content: space-between;
+            align-items: center;
+            margin: 0;
+            padding: var(--nuxeo-collapsible-heading-padding, 4px);
+            background: transparent;
+            border: none;
+            cursor: pointer;
+            text-decoration: none;
+            font-size: 1rem;
+          }
+          /* to fix a blinking default style on safari */
+          button:active {
+            color: inherit;
+          }
+          .heading {
+            width: calc(100% - var(--nuxeo-collapsible-icon-width, 20px));
+            overflow-x: hidden;
+            text-align: start;
+          }
+          iron-icon {
+            --iron-icon-height: var(--nuxeo-collapsible-icon-height, 20px);
+            --iron-icon-width: var(--nuxeo-collapsible-icon-width, 20px);
+          }
+          .start {
+            flex-direction: row-reverse;
+            justify-content: flex-end;
+          }
+          .start-alignment {
+            margin-inline-start: var(--nuxeo-collapsible-icon-width, 20px);
+          }
+          iron-collapse {
+            padding: 0 var(--nuxeo-collapsible-heading-padding, 4px);
+          }
+        </style>
+
+        <button aria-expanded="[[expanded]]" on-tap="_toggle" class$="[[_changeChevronPosition(chevronAtStart)]]">
+          <slot class="heading" name="heading"></slot>
+          <iron-icon icon="[[_toggleIcon(expanded)]]" hidden$="[[!chevron]]"></iron-icon>
+        </button>
+        <iron-collapse
+          opened="[[expanded]]"
+          no-animation="[[!animation]]"
+          class$="[[_alignCollapsedContent(chevronAtStart)]]"
+        >
+          <slot></slot>
+        </iron-collapse>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-collapsible';
+    }
+
+    static get properties() {
+      return {
+        /**
+         * Set animation to false to disable animations
+         */
+        animation: {
+          type: Boolean,
+          value: true,
+        },
+
+        /**
+         * Show or hide the chevron icon
+         */
+        chevron: {
```

Comment:
I'd still rather drop the `chevron` and all the `chevron-*` properties and simply have the chevron as the default header slot content. People not wanting the chevron or wanting a custom chevron can simply override the header content.

Reply:
I understand your point, but, in my opinion, having a collapsible element that works out of the box with a predefined chevron ensures consistency and ease to use. Looking to the main libraries as a reference, they kind of use the same api.